src/components/screens/manage-visitor-session-variables-screen.tsx

```diff
@@ -0,0 +1,732 @@
+import TagListener from "../generic/tag-listener";
+import MqttListener from "../generic/mqtt-listener";
+import Api from "../../api/api";
+import moment from "moment";
+import ElementNavigationPane from "../layouts/element-navigation-pane";
+import produce from "immer";
+import WithDebounce from "../generic/with-debounce";
+import ConfirmDialog from "../generic/confirm-dialog";
+
+/**
+ * Component props
+ */
+interface Props extends WithStyles<typeof styles> {
+  history: History;
+  keycloak: KeycloakInstance;
+  accessToken: AccessToken;
+  exhibitionId: string;
+}
+
+/**
+ * Component state
+ */
+interface State {
+  loading: boolean;
+  error?: Error;
+  foundSessions?: VisitorSession[];
+  foundVisitors?: Visitor[];
+  selectedSession?: VisitorSession;
+  variableDataList?: SessionVariableData[];
+  selectedVariableData?: SessionVariableData;
+  languages: string[];
+  dataChanged: boolean;
+  confirmEmptyOpen: boolean;
+}
+
+interface SessionVariableData {
+  sessionVariable: VisitorSessionVariable;
+  visitorVariable: VisitorVariable;
+}
+
+/**
+ * Component for manage visitor session variables screen
+ */
+export class ManageVisitorSessionVariablesScreen extends React.Component<Props, State> {
+
+  /**
+   * Constructor
+   *
+   * @param props component properties
+   */
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      loading: false,
+      languages: [],
+      dataChanged: false,
+      confirmEmptyOpen: false
+    };
+  }
+
+  /**
+   * Component render method
+   */
+  public render = () => {
+    const { history, keycloak } = this.props;
+    const { error, dataChanged } = this.state;
+
+    return (
+      <BasicLayout
+        history={ history }
+        title={ strings.manageVisitorSessionVariables.title }
+        breadcrumbs={ [] }
+        actionBarButtons={ this.getActionButtons() }
+        keycloak={ keycloak }
+        error={ error }
+        clearError={ () => this.setState({ error: undefined }) }
+        dataChanged={ dataChanged }
+        openDataChangedPrompt
+        noTabs={ true }
+      >
+        { this.resolveComponentToRender() }
+      </BasicLayout>
+    );
+  }
+
+  /**
+   * Resolves what component to render inside basic layout
+   */
+  private resolveComponentToRender = () => {
+    const { classes } = this.props;
+    const { loading, foundSessions, selectedSession } = this.state;
+
+    if (loading) {
+      return (
+        <div className={ classes.loader }>
+          <CircularProgress size={ 50 } color="secondary"></CircularProgress>
+        </div>
+      );
+    }
+
+    if (!foundSessions) {
+      return this.renderTagListener();
+    }
+
+    if (!selectedSession) {
+      return this.renderSessionList();
+    }
+
+    return this.renderSessionVariablesEditor();
+  }
+
+  /**
+   * Renders tag listener
+   */
+  private renderTagListener = () => {
+    const antenna = process.env.REACT_APP_RESET_VISITOR_VARIABLES_ANTENNA;
```

Comment:
If REACT_APP_RESET_VISITOR_VARIABLES_ANTENNA would be to const before the class, it would be easier to know that this component requires external configuration

e.g. `const { REACT_APP_RESET_VISITOR_VARIABLES_ANTENNA } = process.env;`